design_docs/selfhealing_featuretoggle.md

```diff
@@ -0,0 +1,160 @@
+
+# Keptn Feature-Toggle Self-Healing
+
+This document describe how Keptn leverages feature toggle frameworks to enable self-healing for applications by automatically toggling feature toggles in response to problem tickets that are reported to Keptn.
+
+At first, we describe how to write remediation actions agnostic to the actual feature toggle framework used. Secondly, we describe how a user would work with Keptn as a control plane to toggle feature flags for automated operations of the users applications. 
+Please note that in the current version, Keptn will provide built-in functionality for the open-source feature toggle framework [Unleash](https://unleash.github.io/), with other feature toggle platforms to follow.
+
+
+# Prerequisites
+
+
+
+*   Running Keptn Installation
+*   Running [Unleash](https://unleash.github.io/) server as the feature toggle framework
+*   Application with feature flags connected to the Unleash server
+*   Remediation file with feature flags to toggle in case of issues
+*   Running monitoring solution for the application (right now Dynatrace is supported)
+
+
+# Remediation file
+
+The remediation file is necessary to define which remediation action should be taken in response to a problem ticket that is sent to Keptn. Keptn’s internal remediation service will then determine the actions to take and trigger them accordingly. 
+
+Here’s an example of a remediation file:
+
+remediation.yaml
+
+```yaml
+remediations:
+- name: "responsetime_p90_sockshop_carts"
```

Comment:
Would we need to call that `responsetime_p90_sockshop_carts`? The location of the file in the configuration tree should give information about the project and the service. We could then reference the SLIs at that point.

Reply:
correct. having the information in the alert which service in which stage was affected we could map it and won't need this information here.